ibis/common/collections.py

```diff
@@ -276,42 +275,26 @@ def isdisjoint(self, other: collections.abc.Mapping) -> bool:
 
 
 @public
-class FrozenDict(Mapping[K, V], Hashable):
-    """Immutable dictionary with a precomputed hash value."""
-
-    __slots__ = ("__view__", "__precomputed_hash__")
-    __view__: MappingProxyType
+class FrozenDict(dict, Mapping[K, V], Hashable):
+    __slots__ = ("__precomputed_hash__",)
     __precomputed_hash__: int
 
     def __init__(self, *args, **kwargs):
-        dictview = MappingProxyType(dict(*args, **kwargs))
-        dicthash = hash(tuple(dictview.items()))
-        object.__setattr__(self, "__view__", dictview)
-        object.__setattr__(self, "__precomputed_hash__", dicthash)
+        super().__init__(*args, **kwargs)
+        hashable = tuple(sorted(dict(self).items()))
```

Comment:
Could we use `frozenset(self.items())` here?

Reply:
```suggestion
        hashable = frozenset(dict(self).items())
```

I'll add this suggestion to a batch and commit it. We can always revert it before merge if it causes issues.